Nexogen.Libraries.Metrics.Prometheus.Grpc/MetricsBase.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Linq;
+using Grpc.Core;
+
+namespace Nexogen.Libraries.Metrics.Prometheus.Grpc
+{
+    /// <summary>
+    /// Common base class for gRPC-related metrics.
+    /// </summary>
+    public abstract class MetricsBase
+    {
+        /// <summary>
+        /// Registers gRPC metrics.
+        /// </summary>
+        /// <param name="metrics">Builder to register the metrics in.</param>
+        /// <param name="kind">The kind of metrics: server or client.</param>
+        protected MetricsBase(IMetrics metrics, string kind)
+        {
+            string[] labelNames = {"grpc_type", "grpc_service", "grpc_method"};
+
+            started = metrics
+                .Counter()
+                .Name($"grpc_{kind}_started_total")
+                .Help($"Total number of RPCs started on the {kind}.")
+                .LabelNames(labelNames)
+                .Register();
+
+            handled = metrics
+                .Counter()
+                .Name($"grpc_{kind}_handled_total")
+                .Help($"Total number of RPCs completed on the {kind}, regardless of success or failure.")
+                .LabelNames(labelNames.Append("grpc_code").ToArray())
```

Comment:
I would just have this also as a static list.

Reply:
Done.